internal/node/launcher.sh

```diff
@@ -291,6 +271,33 @@ if [[ -n "${BAZEL_NODE_MODULES_ROOT:-}" ]]; then
   fi
 fi
 
+if [ "$PATCH_REQUIRE" = true ]; then
+  require_patch_script=${BAZEL_NODE_PATCH_REQUIRE}
+  # See comment above
+  case "${require_patch_script}" in
+    # Absolute path on unix
+    /*          ) ;;
+    # Absolute path on Windows, e.g. C:/path/to/thing
+    [a-zA-Z]:/* ) ;;
+    # Otherwise it needs to be made relative
+    *           ) require_patch_script="./${require_patch_script}" ;;
+  esac
+  LAUNCHER_NODE_OPTIONS+=( "--require" "$require_patch_script" )
+  # Change the entry point to be the loader.js script so we run code before node
+  MAIN=$(rlocation "TEMPLATED_loader_script")
+else
+  # Entry point is the user-supplied script
+  MAIN=TEMPLATED_entry_point_execroot_path  
+  # TODO: after we link-all-bins we should not need this extra lookup
+  if [[ ! -f "$MAIN" ]]; then
+    if [ "$FROM_EXECROOT" = true ]; then
+      MAIN="$EXECROOT/$MAIN"
+    else
+      MAIN=TEMPLATED_entry_point_manifest_path
```

Comment:
That would be `_nodejs_binary_impl`. I wanted to reuse the logic to detect the execroot path, which is in the launcher. I'm not sure if there's another way to calculate it, or if it's even possible to move that logic to the rule implementation, since it seems to be based around detecting whether some paths exist.

Reply:
@alexeagle any further comment here?

Reply:
yeah I think that makes sense.